src/protocols/secure_channel/CASESession.cpp

```diff
@@ -423,31 +434,44 @@ CHIP_ERROR CASESession::HandleSigmaR1(System::PacketBufferHandle & msg)
     err = tlvReader.EnterContainer(containerType);
     SuccessOrExit(err);
 
+    err = tlvReader.FindElementWithTag(CASETLVTag::kRandom, suppTlvReader);
+    SuccessOrExit(err);
+    VerifyOrExit(kSigmaParamRandomNumberSize == suppTlvReader.GetLength(), err = CHIP_ERROR_INVALID_TLV_ELEMENT);
+    VerifyOrExit(suppTlvReader.GetType() == TLV::kTLVType_ByteString, err = CHIP_ERROR_WRONG_TLV_TYPE);
+    err = suppTlvReader.GetBytes(initiatorRandom, sizeof(initiatorRandom));
+    SuccessOrExit(err);
+
     err = tlvReader.FindElementWithTag(CASETLVTag::kSessionID, suppTlvReader);
     SuccessOrExit(err);
     err = suppTlvReader.Get(encryptionKeyId);
     SuccessOrExit(err);
 
-    err = tlvReader.FindElementWithTag(CASETLVTag::kNumberofTrustedRootIDs, suppTlvReader);
+    ChipLogDetail(SecureChannel, "Peer assigned session key ID %d", encryptionKeyId);
+    mConnectionState.SetPeerKeyID(encryptionKeyId);
+
+    {
+        MutableByteSpan ipkSpan(mIPK, sizeof(mIPK));
+        SuccessOrExit(err = ComputeIPK(mConnectionState.GetPeerKeyID(), ipkSpan));
+    }
```

Comment:
There must be a "match loop" that includes trying every IPK still valid for the given fabric (there may be more than one). We do not have that yet (Group Key Management Cluster) to spec in here, so I suggest adding a TODO to iterate over all candidate IPKs for each fabric.

Reply:
Updated.